src/ContentRepository/FileBase.cs

```diff
@@ -140,5 +142,26 @@ private static void AssertExecutableType(Node node)
                     throw new InvalidContentException(SR.GetString(SR.Exceptions.File.Error_ForbiddenExecutable)); 
             }
 	    }
+
+	    private static void SetBinaryData(FileBase file)
+	    {
+	        var binaryData = file.Binary;
+
+	        var mimeTypeOriginal = binaryData.ContentType;
+
+	        if (string.IsNullOrEmpty(binaryData.FileName))
+	            binaryData.FileName = file.Name;

```

Comment:
Maybe this can be simplified because setting the FileName sets the ContentType automatically, we do not have to set it manually below.

Reply:
```csharp
if (string.IsNullOrEmpty(binaryData.FileName))
   binaryData.FileName = file.Name;

if (!string.IsNullOrEmpty(mimeTypeOriginal))
   binaryData.ContentType = mimeTypeOriginal;
```